File: tests/fixtures/patterns-positive/decorator/mutB.ts
// POSITIVO (mutación B: Decorator aplanado; acumulador con 3 capas condicionales).
function describeShape(shape: Shape, hasColor: boolean, hasBorder: boolean, hasShadow: boolean): string {
  let result = shape.describe();
  if (hasColor) {
    result = "colored(" + result + ")";
  }
  if (hasBorder) {
    result = "bordered(" + result + ")";
  }
  if (hasShadow) {
    result = "shadowed(" + result + ")";
  }
  return result;
}

File: tests/fixtures/patterns-positive/decorator/mutB.py
# POSITIVO (mutación B: Decorator aplanado; acumulador con 3 capas condicionales).
def describe_shape(shape, has_color, has_border, has_shadow):
    result = shape.describe()
    if has_color:
        result = "colored(" + result + ")"
    if has_border:
        result = "bordered(" + result + ")"
    if has_shadow:
        result = "shadowed(" + result + ")"
    return result
